src/main/scala/mesosphere/marathon/core/launchqueue/impl/TaskLauncherActor.scala

```diff
@@ -278,7 +278,9 @@ private class TaskLauncherActor(
 
         // Only instances that scheduled or provisioned have not seen a Mesos update. The provision timeouts waits for
         // any Mesos update. Thus we can safely kill the provision timeout in all other cases, even on a TASK_FAILED.
-        if (!instance.isProvisioned && !instance.isScheduled) {
+        // with stable ids, TASK_FAILED ends up yielding instance.isScheduled typically in case of goal: Running
+        // because of that we have to handle instance becoming terminal explicitly
+        if (!instance.isProvisioned && provisionTimeouts.contains(instanceId)) {
```

Comment:
```suggestion
        if (!instance.isProvisioned) {
```
I think `provisionTimeouts.get(instanceId).foreach(_.cancel())` already does the check 🙂